backend/README.md

```diff
@@ -0,0 +1,137 @@
+
+#### get
+The get function acts like before and fetches the server as an object from the database. This is then returned as a JSON object.
+
+#### delete
+This function deletes the specific server that was passed in. It will return a code 204 is this completes correctly.
+
+#### put
+This function will update a server given it's id.
+
+#### request
+This function handles routing a request to the actual controller residing at the address contained in the server object. A PAYLOAD is defined that takes in the type of request, the endpoint it is going to, and any optional payload in raw form. This payload is sent to the server that is returned from the query (that returns the ONE object after filtering based on id). It will return 404 if the server is not able to be contacted or does not exist. This function utilizes a helper function "route_requset" that handles the actual setup of each request based on the type of request.
+
+## Development Setup
+### Linux
+Make sure you have Python 3.6 installed.
+
+```sh
+sudo add-apt-repository ppa:deadsnakes/ppa
+sudo apt update
+sudo apt install python3.6
+```
+
+Make sure you have pipenv installed through pip.
+```bash
+pip install pipenv
+```
+
+Make sure your cd is `backend` directory.
+
+Now run:
+```bash
+pipenv install
+```
+This will create a virtualenv with python3
+
+Finally run:
+```bash
+pipenv shell
+```
+This enables the environment with the lock file already included in the repository.
+
+### Windows
+Setup is very different for windows. You must first download python 3.6 from their website.
+
+Next download [get-pip.py](https://bootstrap.pypa.io/get-pip.py)
+Execute it with console by pressing windows key, type in cmd, then press enter. Run python get-pip.py from where you downloaded it to. You may have to put python in your PATH (Google if needed).
+
+Install pipenv through pip:
+```bash
+pip install pipenv
+```
+This will create a virtualenv with python3
+
+Make sure your cd is `backend` directory.
+
+Install the dependencies of the project:
+```bash
+pipenv install
+```
+
+Finally run:
+```bash
+pipenv shell
+```
+This enables the environment with the lock file already included in the repository.
+
+## Server Setup
+### Linux
+Assuming the environment is enabled (pipenv shell) in the backend directory, you may now start the server in development mode with:
+```bash
+python application.py dev
+```
+
+### Windows
+Assuming the environment is enabled (pipenv shell) in the backend directory, you may now start the server with:
+```bash
+python application.py dev
+```
+
+## Database Setup
+### Docker
+First install Docker. For Ubuntu you do;
```

Comment:
We can look into an automated setup script. We didn't make it a priority since we figured that the database will be hosted on an actual server in the future and only a couple people will ever need to host a local version (mostly just during this initial setup). We will definitely take a look at making it simpler.

Reply:
Hmm okay that sounds fair. But still it's pretty easy to put the few commands in a start_server script or whatever. No hurry though